BUILD

```diff
@@ -0,0 +1,308 @@
+        "@com_google_googletest//:gtest_main",
+        ":raylet_lib",
+    ],
+)
+
+cc_test(
+    name = "logging_test",
+    srcs = ["src/ray/util/logging_test.cc"],
+    deps = [
+        "@com_google_googletest//:gtest_main",
+        ":ray_util",
+    ],
+)
+
+cc_test(
+    name = "task_dependency_manager_test",
+    srcs = ["src/ray/raylet/task_dependency_manager_test.cc"],
+    deps = [
+        "@com_google_googletest//:gtest_main",
+        ":raylet_lib"
+    ],
+)
+
+cc_test(
+    name = "task_test",
+    srcs = ["src/ray/raylet/task_test.cc"],
+    deps = [
+        "@com_google_googletest//:gtest_main",
+        ":raylet_lib",
+    ],
+)
+
+cc_library(
+    name = "object_manager",
+    srcs = glob([
+        "src/ray/object_manager/*.cc",
+    ]),
+    hdrs = glob([
+        "src/ray/object_manager/*.h",
+    ]),
+    includes = [
+        "src",
+    ],
+    deps = [
+        ":gcs",
+        ":object_manager_fbs",
+        ":ray_common",
+        ":ray_util",
+        "@boost//:asio",
+        "@plasma"
+    ],
+)
+
+cc_binary(
+    name = "object_manager_test",
+    testonly = 1,
+    srcs = ["src/ray/object_manager/test/object_manager_test.cc"],
+    deps = [
+        "@com_google_googletest//:gtest_main",
+        ":object_manager",
+    ],
+)
+
+cc_binary(
+    name = "object_manager_stress_test",
+    testonly = 1,
+    srcs = ["src/ray/object_manager/test/object_manager_stress_test.cc"],
+    deps = [
+        "@com_google_googletest//:gtest_main",
+        ":object_manager",
+    ],
+)
+
+cc_library(
+    name = "ray_util",
+    srcs = glob(
+        [
+            "src/ray/*.cc",
+            "src/ray/util/*.cc",
+        ],
+        exclude = [
+            "src/ray/util/logging_test.cc",
+            "src/ray/util/signal_test.cc"
+        ],
+    ),
+    hdrs = glob([
+        "src/ray/*.h",
+        "src/ray/util/*.h",
+    ]),
+    includes = [
+        "src",
+    ],
+    deps = [
+      "@plasma",
+      ":sha256"
+    ]
+)
+
+cc_library(
+    name = "ray_common",
+    srcs = ["src/ray/common/client_connection.cc",
+          "src/ray/common/common_protocol.cc"],
+    hdrs = ["src/ray/common/client_connection.h",
+          "src/ray/common/common_protocol.h"],
+    includes = [
+        "src/ray/gcs/format",
+    ],
+    deps = [
+        ":gcs_fbs",
+        ":node_manager_fbs",
+        ":ray_util",
+        "@boost//:asio",
+    ],
+)
+
+cc_library(
+    name = "sha256",
+    srcs = [
+        "src/ray/thirdparty/sha256.c",
+    ],
+    hdrs = [
+        "src/ray/thirdparty/sha256.h",
+    ],
+    includes = ["src/ray/thirdparty"],
+)
+
+cc_library(
+    name = "hiredis",
+    srcs = glob([
+      "src/ray/thirdparty/ae/ae.c",
+      "src/ray/thirdparty/hiredis/*.c",
+    ]),
+    hdrs = glob([
+      "src/ray/thirdparty/ae/*.h",
+      "src/ray/thirdparty/hiredis/*.h",
+      "src/ray/thirdparty/hiredis/adapters/*.h",
+      "src/ray/thirdparty/hiredis/dict.c",
+      "src/ray/thirdparty/ae/ae_kqueue.c"
```

Comment:
@ericl What's the best way to do this in a platform independent way? On linux we need qe_epoll.c here.